am-lib/src/test/java/uk/gov/hmcts/reform/amlib/internal/utils/SecurityClassificationTest.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.amlib.internal.utils;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import uk.gov.hmcts.reform.amlib.enums.SecurityClassification;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.amlib.enums.SecurityClassification.NONE;
+import static uk.gov.hmcts.reform.amlib.enums.SecurityClassification.PRIVATE;
+import static uk.gov.hmcts.reform.amlib.enums.SecurityClassification.PUBLIC;
+import static uk.gov.hmcts.reform.amlib.enums.SecurityClassification.RESTRICTED;
+
+@SuppressWarnings("PMD.UnusedPrivateMethod")
+class SecurityClassificationTest {
+
+    @ParameterizedTest
+    @MethodSource("createArguments")
+    void test(Arguments args) {
+        boolean isVisible = args.resourceClassification.isVisible(args.roleClassification.getHierarchy());
+        assertThat(isVisible).isEqualTo(args.expectedResult);
+    }
+
+    private static Stream<Arguments> createArguments() {
+        return Stream.of(
+            new Arguments(true, PUBLIC, NONE),
+            new Arguments(true, PUBLIC, PUBLIC),
+            new Arguments(false, PUBLIC, PRIVATE),
+            new Arguments(false, PUBLIC, RESTRICTED),
+            new Arguments(true, PRIVATE, NONE),
+            new Arguments(true, PRIVATE, PUBLIC),
+            new Arguments(true, PRIVATE, PRIVATE),
+            new Arguments(false, PRIVATE, RESTRICTED),
+            new Arguments(true, RESTRICTED, NONE),
+            new Arguments(true, RESTRICTED, PUBLIC),
+            new Arguments(true, RESTRICTED, PRIVATE),
+            new Arguments(true, RESTRICTED, RESTRICTED),
+            new Arguments(false, NONE, PUBLIC),
+            new Arguments(false, NONE, PRIVATE),
+            new Arguments(false, NONE, RESTRICTED)
+        );
+    }
+
+    private static class Arguments {
```

Comment:
Istead of custom data class you might use `org.junit.jupiter.params.provider.Arguments`. 

Your provider would change into:

```
private static Stream<Arguments> createArguments() {
    return Stream.of(
        Arguments.of(true, PUBLIC, NONE)
    );
}
```

and your test method would become:

```
@ParameterizedTest
@MethodSource("createArguments")
void test(boolean expectedResult, SecurityClassification roleClassification, SecurityClassification resourceClassification) {}
```

Method should be renamed as well to express expectation.

Reply:
Another suggestion could be splitting into 2 test methods, one covering `true` (visible) scenarios, other one covering `false` (invisible) scenarios

Reply:
On second though I am not sure how much that split would improve. On one hand it would remove that nasty true/false that is cannot be implicitly named in places that uses it, on the other hand order of combinations clearly shows where visibility boundary is.

Reply:
updated - less code. I like.